torch/_inductor/dependencies.py

```diff
@@ -359,5 +359,44 @@ def extract_read_writes(
     )
 
 
+def extract_input_node_reduction_ranges(
+    input_node: ".ir.TensorBox",
+) -> Tuple[Optional[List[sympy.Expr]], Optional[List[sympy.Expr]]]:
+    """
+    Returns the size and reduction size of all inputs, if the sizes and reduction_sizes (if exist) are all the same.
+    It's possible that a node has multiple inputs, some are Reduction nodes and others are Pointwise nodes.
+    In this case, reduction_sizes of the Reduction nodes need to be the same.
+    Otherwise returns (None, None).
+    """
+
+    from .ir import ComputedBuffer, Loops
+    if not isinstance(input_node.data.data, Loops):
+        # Input node has already been realized. Return its size and reduction_size.
+        return input_node.get_size(), input_node.get_reduction_size()
+
+    # This is one issue: what if there are permutations between the input node and its dependent realized nodes?
```

Comment:
@jansel Wonder do you have any suggestions for this?

Reply:
In addition to permutations there are views which change the ndimension.

Is it ok if this function is approximate?  Or are there correctness issues if it is wrong?

Reply:
Using reduction_sizes from dependent nodes have a better chance to fuse these nodes.

e.g. The current case is:
```
x1 = layer_norm(x0)
x2 = amax(x1)
x3 = to_fp8(x1)
```

Inductor generates these nodes:
```
n0=WelfordReduction()
n1=WelfordReduction()
n2=WelfordReduction()
n3=Pointwise()
n4=Reduction()
n5=Pointwise()
```

Currently n0, n1, n2, n3, n5 are fused together. n3, n4 are fused together.
I'd like to make first level reduction ranges of n4 the same as n0 / n1 / n2, so that n0, n1, n2, n3, first level n4, n5 can be fused together.

So it seem to me that we cannot use approximate values here for n4 reduction sizes.